### PR TITLE
Added Postable class

### PR DESCRIPTION
**Pull request by jackietsay**
Now User, Event, Page, and Group are postable objects. Each post has to reference a postedObj which represents where the post is posted. For wall posts, the postedObj is the user itself since we don't have a wall or news object.

### CODE REVIEW

**Comment by amshali** [comment]
The Postable name is confusing. We need to find a better name for it.

**Comment by jackietsay** [comment]
It's exactly the same as commentable. What better name could there be?

On Apr 18, 2012 9:03 AM, "Amin Shali" <
reply@reply.github.com>
wrote:

The Postable name is confusing. We need to find a better name for it.

---

File: app/controllers/Posts.java [hunk 1]
```diff
@@ -60,9 +55,9 @@ public static void makeNewPagePost(String postContent, String pid) {
     renderTemplate(m);
   }
 
-  public static void poke (Long userId) {
+  public static void poke(Long userId) {
     String poked = new String(user() + " has poked " + (User)User.findById(userId) + "!");
-    new Post(user(), new Date().toString(), poked).save();
+    new Post(user(), user(), poked).save();
```
Comment (amshali) [inline app/controllers/Posts.java:60]: What is the second parameter for?

File: app/models/User.java [hunk 2]
```diff
@@ -128,24 +128,20 @@ public static User getUser(String login) {
     return Message.find("SELECT m FROM Message m WHERE m.author = ?1 OR m.recipient = ?1", this).fetch();
   }
 
-  public List<Post> posts() {
```
Comment (amshali) [inline app/models/User.java:131]: Did you change the reference to this method everywhere?

Comment (jackietsay) [inline app/models/User.java:131]: Yup!

File: app/controllers/Posts.java (hunk 1 above)
Comment (jackietsay) [inline app/controllers/Posts.java:60]: The first parameter is the postedObj and the second is the author. In the case of posting on a user's wall, they're both the same. I wasn't sure here whether a poked post should appear on the poker's wall or the pokee.

Comment (amshali) [inline app/controllers/Posts.java:60]: okay, I have to merge this manually.

**Comment by amshali** [comment]
You forgot to add the Postable class. Please add it.

**Pull request merged**